docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,59 @@
+# PL/I Language Server Architecture
+
+This document describes the architecture of the PL/I language server (LS).
+
+## Source Files
+
+The core of all data are the so called `SourceFile` objects.
```

Comment:
Can it happen that a file can be included but at the same time can also be an entry point?

Reply:
Not really. The `uri` should point to the top most element (topologically sorted) of the include tree :)